dao/src/main/java/org/thingsboard/server/dao/notification/DefaultNotificationSettingsService.java

```diff
@@ -81,6 +91,53 @@ public NotificationSettings findNotificationSettings(TenantId tenantId) {
                 });
     }
 
+    @Override
+    public UserNotificationSettings saveUserNotificationSettings(TenantId tenantId, UserId userId, UserNotificationSettings settings) {
+        UserSettings userSettings = new UserSettings();
+        userSettings.setUserId(userId);
+        userSettings.setType(UserSettingsType.NOTIFICATIONS);
+        userSettings.setSettings(JacksonUtil.valueToTree(settings));
+        userSettingsService.saveUserSettings(tenantId, userSettings);
+        return formatUserNotificationSettings(settings);
+    }
+
+    @Override
+    public UserNotificationSettings getUserNotificationSettings(TenantId tenantId, UserId userId, boolean format) {
+        UserSettings userSettings = userSettingsService.findUserSettings(tenantId, userId, UserSettingsType.NOTIFICATIONS);
+        UserNotificationSettings settings;
+        if (userSettings != null) {
+            settings = JacksonUtil.treeToValue(userSettings.getSettings(), UserNotificationSettings.class);
```

Comment:
What if we change the structure of UserNotificationSettings? I would recommend adding try catch and fallback to the UserNotificationSettings.DEFAULT.